source/constructs/api/data_source/service.py

```diff
@@ -2026,38 +2026,32 @@ def list_data_location():
     for item in provider_list:
         regions = crud.list_distinct_region_by_provider(item.id)
         if not regions:
-            location = DataLocationInfo()
-            location.source = item.provider_name
-            location.region = None
-            location.coordinate = None
-            location.account_count = 0
-            res.append(location)
+            # location = DataLocationInfo()
+            # location.source = item.provider_name
+            # location.region = None
+            # location.coordinate = None
+            # location.account_count = 0
+            # res.append(location)
             continue
         for subItem in regions:
+            accounts = crud.list_account_by_provider_and_region(item.id, subItem.region_name)
+            if len(accounts) == 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)